scripts/module/weapon_pickups.nut

```diff
@@ -18,19 +18,18 @@ function onPlayerPart( player, reason ) {
 function onPlayerHealthChange( player, lastHP, newHP ) {
   if ( newHP < 0.1 && alive[player.ID]) {
     alive[player.ID] = false;
-    local pos = player.Pos;
-    local xOffset = 0;
-    local zOffset = 0;
+    local pos = player.Pos,
+      x, y, alpha;
+    local radius = 1.0;
     local pickup = null;
     for (local i = 0; i <= 8; ++i) {
       if ( player.GetWeaponAtSlot(i) ) {
+        alpha = i / 8.0 * 2.0 * PI;
```

Comment:
I'd add the player angle or a random jitter, otherwise it's painfully obvious that certain weapons always spawn facing north

Reply:
It will probably lead to ugly pickup distribution (where it looks like parallel rows of weapons), and might also lead to certain weapons always spawning in awkward places (imagine players standing infront of a small wall to shoot, then the same kind of gun might always spawn inside the wall *every time*). By adding jitter or natural randomness this can be prevented.